### PR TITLE
Ticket 5

### PR DESCRIPTION
**Pull request by TheRealImrano**
submitting ticket-5 for review. Test has .only because ticket-5 was started before ticket-4 was merged into main, as it was awaiting review. So, tests for GET /api/articles/:article_id wont run until after merge as that code is missing from the ticket (previous ticket was proof of it working however).

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -51,4 +51,76 @@ describe('api; GET /api', ()=>{
...
+            return request(app)
+            .get('/api/articles')
+            .expect(200)
+            .then((response)=>{
+                const {articles} = response.body;
+                const expectedProperties = ['article_id', 'title', 'topic', 'author', 'comment_count', 'created_at', 'votes', 'article_img_url'];
 
+                expect(articles).toHaveLength(13);
+                articles.forEach((article)=>{
+                    expect(Object.keys(article)).toHaveLength(8);
```
Comment (nc-poppy-m) [inline __tests__/app.test.js:99]: interesting way of doing things! Look into .toMatchObject which might make your life a little easier

File: __tests__/app.test.js [hunk 2]
```diff
@@ -51,4 +51,76 @@ describe('api; GET /api', ()=>{
...
+        })
+        test('data returned from endpoint has articles listed by date in descending order', ()=>{
+            return request(app)
+            .get('/api/articles')
+            .expect(200)
+            .then((response)=>{
+                const {articles} = response.body;
+
+                for (let i = 0; i < articles.length - 1; i++) {
+                    const currentArticleDate = new Date(articles[i].created_at);
```
Comment (nc-poppy-m) [inline __tests__/app.test.js]: This works but is quite a complicated solution - try to use ```jest sorted``` to make this easier for you. https://www.npmjs.com/package/jest-sorted

File: app.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 const express = require("express");
 const { getTopics } = require("./controllers/topicsController");
 const { getEndpoints } = require("./controllers/apiController");
+const { getAllArticles } = require("./controllers/articleController");
 
 
 const app = express();
```
Comment (nc-poppy-m) [inline app.js:7]: Remove app.use(express.json()) until you need it

**Review by nc-poppy-m — APPROVED** [review]
Good work Ali! Going to approve it but I've left some feedback as suggestions. The only mandatory thing is to remove app.use(express.json()) as we don't need this until a later task. 

As a side note, we usually can't accept prs with .only even if they worked in previous tests as something may have gone wrong (it's only because I feel confident about your work here), so make sure your work is sorted in a way that you don't have tests for unimplemented functionality on a new branch 🙂

**Comment by TheRealImrano** [comment]
Thanks for the feedback Poppy! I'll implement these changes asap, and definitely look into the suggested jest matchers also.

**Review by TheRealImrano — COMMENTED** [review]
ticket-5 ready to merge

**Pull request merged**